src/gt4py/next/iterator/ir_utils/misc.py

```diff
@@ -71,3 +71,79 @@ def is_equal(a: itir.Expr, b: itir.Expr):
     return a == b or (
         CannonicalizeBoundSymbolNames.apply(a) == CannonicalizeBoundSymbolNames.apply(b)
     )
+
+
+def canonicalize_as_fieldop(expr: itir.FunCall) -> itir.FunCall:
+    """
+    Canonicalize applied `as_fieldop`s.
+
+    In case the stencil argument is a `deref` wrap it into a lambda such that we have a unified
+    format to work with (e.g. each parameter has a name without the need to special case).
+    """
+    assert cpm.is_applied_as_fieldop(expr)
+
+    stencil = expr.fun.args[0]  # type: ignore[attr-defined]
+    domain = expr.fun.args[1] if len(expr.fun.args) > 1 else None  # type: ignore[attr-defined]
+    if cpm.is_ref_to(stencil, "deref"):
+        stencil = im.lambda_("arg")(im.deref("arg"))
+        new_expr = im.as_fieldop(stencil, domain)(*expr.args)
+
+        return new_expr
+
+    return expr
+
+
+def _remove_let_alias(let_expr: itir.FunCall):
+    assert cpm.is_let(let_expr)
+    is_aliased_let = True
+    for param, arg in zip(let_expr.fun.params, let_expr.args, strict=True):  # type: ignore[attr-defined]  # ensured by cpm.is_let
+        is_aliased_let &= cpm.is_ref_to(arg, param.id)
+    if is_aliased_let:
+        return let_expr.fun.expr  # type: ignore[attr-defined]  # ensured by cpm.is_let
+    return let_expr
+
+
+def unwrap_scan(stencil: itir.Lambda | itir.FunCall):
```

Comment:
This function has been moved, the only addition is the `_remove_let_alias` call.